src/ConnectedMode.UnitTests/Suppressions/ServerIssuesStoreTests.cs

```diff
@@ -30,6 +34,84 @@ public class ServerIssuesStoreTests
         {
             MefTestHelpers.CheckTypeCanBeImported<ServerIssuesStore, IServerIssuesStore>();
+            MefTestHelpers.CheckTypeCanBeImported<ServerIssuesStore, IServerIssuesStoreWriter>();
+        }
+
+        [TestMethod]
+        public void AddIssues_ListOfIssuesAsExpected()
+        {
+            var issue1 = CreateIssue("1", false);
+            var issue2 = CreateIssue("2", false);
+            var issue3 = CreateIssue("3", false);
+
+            var testSubject = new ServerIssuesStore();
+
+            var result = testSubject.Get();
+            result.Count().Should().Be(0);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1, issue2 }, clearAllExistingIssues: false);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(2);
+            result.Should().Contain(issue1);
+            result.Should().Contain(issue2);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue3 }, clearAllExistingIssues: true);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(1);
+            result.Should().Contain(issue3);
+        }
+
+        [TestMethod]
+        public void UpdateIssues_ListOfIssuesAsExpected()
+        {
+            var issue1 = CreateIssue("issue1", true);
+            var issue2 = CreateIssue("issue2", false);
+
+            var testSubject = new ServerIssuesStore();
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1, issue2 }, clearAllExistingIssues: false);
+
+            var result = testSubject.Get();
+            result.Count().Should().Be(2);
+            result.Should().Contain(issue1);
+            result.Should().Contain(issue2);
+            issue1.IsResolved.Should().BeTrue();
+            issue2.IsResolved.Should().BeFalse();
+
+            testSubject.UpdateIssue("issue1", false);
+            issue1.IsResolved.Should().BeFalse();
+
+            testSubject.UpdateIssue("issue2", true);
+            issue2.IsResolved.Should().BeTrue();
+
+            // This test is to insure it doesn't just flip the state.
+            testSubject.UpdateIssue("issue2", true);
+            issue2.IsResolved.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void AddOrUpdateIssues_EventIsInvoked()
+        {
+            var issue1 = CreateIssue("issue1", true);
+
+            var testSubject = new ServerIssuesStore();
+            var eventMock = new Mock<EventHandler>();
+            testSubject.ServerIssuesChanged += eventMock.Object;
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1 }, clearAllExistingIssues: false);
+            eventMock.Verify(x => x(testSubject, EventArgs.Empty), Times.Once);
+
+            testSubject.UpdateIssue("issue1", false);
+            eventMock.Verify(x => x(testSubject, EventArgs.Empty), Times.Exactly(2));
+        }
+
+        private static SonarQubeIssue CreateIssue(string key, bool isResolved)
```

Comment:
Is there a reason why we dont have an interface to mock this?

Reply:
Not really. It's a data-only class so there was less need to i.e. no functionality to mock, so I guess we never got round to it.